prov/rxm/src/rxm_cq.c

```diff
@@ -228,29 +228,20 @@ static inline int rxm_finish_sar_segment_send(struct rxm_tx_buf *tx_buf)
 
 	rxm_tx_buf_release(tx_entry->ep, tx_buf);
 	/* If `segs_left` == 0, all segments of the message have been fully sent */
-	if (!--tx_entry->segs_left) {
-		return rxm_finish_send_nobuf(tx_entry);
-	} else if (!dlist_empty(&tx_entry->deferred_tx_buf_list)) {
-		ssize_t ret = 0;
-
-		while (!dlist_empty(&tx_entry->deferred_tx_buf_list) && !ret) {
-			tx_buf = container_of(tx_entry->deferred_tx_buf_list.next,
-					      struct rxm_tx_buf, hdr.entry);
-			ret = fi_send(tx_entry->conn->msg_ep, &tx_buf->pkt,
-				      sizeof(tx_buf->pkt) + tx_buf->pkt.ctrl_hdr.seg_size,
-				      tx_buf->hdr.desc, 0, tx_buf);
-			if (!ret)
-				dlist_remove(&tx_buf->hdr.entry);
+	if (!(--tx_entry->segs_left - tx_entry->fail_segs_cnt)) {
```

Comment:
`tx_entry->segs_left` is already decremented by `tx_entry->fail_segs_cnt` in `rxm_ep_sar_tx_prepare_and_send_segment`.